File: data/code_pairs/cpp/319.cpp
#include <iostream>
#include <memory>

// Abstract base class
class t {
public:
    virtual void display() = 0; // Pure virtual function for polymorphic behavior
    virtual ~t() = default; // Virtual destructor for proper cleanup
};

// Derived class implementing the display method
class t_derived : public t {
public:
    void display() override {
        std::cout << "Display from derived class." << std::endl;
    }
};

// Test function demonstrating dynamic allocation and polymorphism
void test() {
    std::unique_ptr<t> obj = std::make_unique<t_derived>(); // Dynamic allocation of derived class
    obj->display(); // Polymorphic call to display method
}

int main() {
    test();
    return 0;
}

File: data/code_pairs/fortran/319.f90
module test_mod
  implicit none

  ! Define an abstract type
  type, abstract :: t
  contains
    procedure(poly_interface), deferred :: display
  end type t

  ! Interface for polymorphic behavior
  abstract interface
    subroutine poly_interface(self)
      import :: t
      class(t), intent(in) :: self
    end subroutine poly_interface
  end interface

  ! A derived type extending the abstract type
  type, extends(t) :: t_derived
  contains
    procedure :: display => display_derived
  end type t_derived

contains

  ! Implementation of the derived type's display method
  subroutine display_derived(self)
    class(t_derived), intent(in) :: self
    print *, "Display from derived class."
  end subroutine display_derived

  ! Test subroutine showcasing dynamic allocation and polymorphism
  subroutine test()
    class(t), allocatable :: obj
    type(t_derived) :: derived_obj

    allocate(t_derived :: obj)
    call obj%display()
  end subroutine test

end module test_mod

program main
  use test_mod
  call test()
end program main